aiida/orm/nodes/data/code/containerized.py

```diff
@@ -0,0 +1,116 @@
+        super().__init__(**kwargs)
+        self.engine_command = engine_command
+        self.image_name = image_name
+
+    @property
+    def engine_command(self) -> str:
+        """Return the engine command with image as template field of the containerized code
+
+        :return: The engine command of the containerized code
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_ENGINE_COMMAND)
+
+    @engine_command.setter
+    def engine_command(self, value: str) -> None:
+        """Set the engine command of the containerized code
+
+        :param value: The engine command of the containerized code
+        """
+        type_check(value, str)
+
+        if '{image_name}' not in value:
+            raise ValueError("the '{image_name}' template field should be in engine command.")
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_ENGINE_COMMAND, value)
+
+    @property
+    def image_name(self) -> str:
+        """The image name of container
+
+        :return: The image name of container.
+        """
+        return self.base.attributes.get(self._KEY_ATTRIBUTE_IMAGE_NAME)
+
+    @image_name.setter
+    def image_name(self, value: str) -> None:
+        """Set the image name of container
+
+        :param value: The image name of container.
+        """
+        type_check(value, str)
+
+        self.base.attributes.set(self._KEY_ATTRIBUTE_IMAGE_NAME, value)
+
+    def get_prepend_cmdline_params(
+        self, mpi_args: list[str] | None = None, extra_mpirun_params: list[str] | None = None
+    ) -> list[str]:
+        """Return the list of prepend cmdline params for mpi seeting
+
+        :return: list of prepend cmdline parameters."""
+        engine_cmdline = self.engine_command.format(image_name=self.image_name)
+        engine_cmdline_params = engine_cmdline.split()
+
+        return (mpi_args or []) + (extra_mpirun_params or []) + engine_cmdline_params
+
+    @classmethod
+    def _get_cli_options(cls) -> dict:
+        """Return the CLI options that would allow to create an instance of this class."""
+        options = {
+            'engine_command': {
+                'required':
+                True,
+                'prompt':
+                'Engine command',
+                'help': (
+                    'The command to run the container. It must contain the placeholder '
+                    '{image_name} that will be replaced with the `image_name`.'
+                ),
+                'type':
+                click.STRING,
+            },
+            'image_name': {
+                'required': True,
+                'type': click.STRING,
+                'prompt': 'Image name',
+                'help': 'Name of the image container in which to the run the executable.',
+            },
+        }
+        options.update(**super()._get_cli_options())
+
+        return options
+
+
+class InstalledContainerizedCode(Containerized, InstalledCode):
```

Comment:
I think you may have reviewed an old commit, because this was already removed. It simply is called `ContainerizedCode`. I am not sure how you are still getting this old diff :confused:

Reply:
sorry, I started the review some time ago but then had to stop.
I resolved all of the outdated comments (there was no way to delete them before I submitted the review)